staging/src/k8s.io/pod-security-admission/policy/check_seccomp_restricted.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package policy
+
+import (
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/pod-security-admission/api"
+)
+
+func init() {
+	addCheck(CheckSeccompRestricted)
+}
+
+func CheckSeccompRestricted() Check {
+	return Check{
+		ID:    "seccomp_restricted",
+		Level: api.LevelRestricted,
+		Versions: []VersionedCheck{
+			{
+				MinimumVersion: api.MajorMinorVersion(1, 0),
+				CheckPod:       seccomp_1_0_restricted,
+			},
+			{
+				MinimumVersion: api.MajorMinorVersion(1, 19),
+				CheckPod:       seccomp_1_19_restricted,
+			},
+		},
+	}
+}
+
+// seccomp_1_0_restricted checks restricted policy on seccomp alpha annotation
+func seccomp_1_0_restricted(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec) CheckResult {
+	forbidden := sets.NewString()
+	podSeccompSet := false
+
+	if val, ok := podMetadata.Annotations[annotationKeyPod]; ok {
+		if !validSeccompAnnotation(val) {
+			podAnnotationField := field.NewPath("metadata").Child("annotations", annotationKeyPod)
+			forbidden.Insert(fieldValue(podAnnotationField, val))
+		} else {
+			podSeccompSet = true
+		}
+	}
+
+	visitContainersWithPath(podSpec, field.NewPath("spec"), func(c *corev1.Container, path *field.Path) {
+		annotation := annotationKeyContainerPrefix + c.Name
+		if val, ok := podMetadata.Annotations[annotation]; ok {
+			if val == corev1.SeccompProfileNameUnconfined {
```

Comment:
why not `if !validSeccompAnnotation(val)`? the container could set the annotation value to "" and this would result in the container being unconfined even if the pod set to runtime/default (see kubelet impl in `getSeccompProfilePath`)